moda-contracts/src/CatalogFactory.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+import "./interfaces/ICatalogFactory.sol";
+import "./interfaces/ICatalog.sol";
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+
+contract CatalogFactory is ICatalogFactory {
+    /// Storage Variables
+    address public immutable catalogBeacon;
+    address[] public catalogs;
+
+    /// Errors
+    error AddressCannotBeZero();
+
+    /// Constructor
+    constructor(address catalogBeacon_) {
+        if (catalogBeacon_ == address(0)) {
+            revert AddressCannotBeZero();
+        }
+        catalogBeacon = catalogBeacon_;
+    }
+
+    /// External Functions
+    /**
+     * @inheritdoc ICatalogFactory
+     */
+    function deployCatalog(
+        string calldata catalogName,
+        string calldata chainId,
+        string calldata catalogVersion,
+        address modaRegistry,
+        address splitsFactory,
+        address modaBeneficiary
```

Comment:
`modaTreasury` is a more appropriate term I think, but let's move this and splitsFactory to the ModaRegistry contract. You can create a new interface like `IOfficialContracts` that has `getSplitsFactory => ISplitsFactory`, getModaTreasury => payable address`, `getProfile => IProfile`, etc. This interface can be used inside of Catalog and not take up too much space in the code footprint.

Just add the look up functions you need (no need to add something like `getProfile` now if you aren't using it

The benefit of this is that we can change it in the ModaRegistry and everything else will update.

Reply:
Done.